packages/pwa-storefront-mcp/src/utils/AddComponentTool.js

```diff
@@ -0,0 +1,479 @@
+/*
+ * Copyright (c) 2025, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+export class AddComponentTool {
+    constructor() {
+        this.componentTemplates = {
+            button: this.createButtonComponent,
+            card: this.createCardComponent,
+            modal: this.createModalComponent,
+            form: this.createFormComponent,
+            list: this.createListComponent,
+            header: this.createHeaderComponent,
+            footer: this.createFooterComponent,
+            product: this.createProductComponent,
+            cart: this.createCartComponent
```

Comment:
Are we limited to adding these components only? And what happens if the API contract of these components changes?